execution/scan/vector_selector.go

```diff
@@ -23,9 +23,10 @@ import (
 )
 
 type vectorScanner struct {
-	labels    labels.Labels
-	signature uint64
-	samples   *storage.MemoizedSeriesIterator
+	labels     labels.Labels
+	signature  uint64
+	samples    *storage.MemoizedSeriesIterator
+	filterFunc func(model.StepVector) model.StepVector
```

Comment:
I dont think this is ever populated anywhere right?

Reply:
There might be some gaps in my understanding (sorry) but I thought, the filterFunc field within the vectorScanner struct is being populated within the Next method of the vectorSelector struct. The Next method creates a new vectorScanner and assigns the filterFunc to it before returning the result. After the filterFunc is assigned within the Next method of the vectorSelector struct, the actual filtering process occurs when iterating through the vectors and applying the filterFunc to each StepVector. When the Next method is called, it returns a slice of model.StepVector instances. Each model.StepVector in the returned slice is processed by the filterFunc that was assigned to the corresponding vectorScanner during its creation. The filterFunc is applied to each model.StepVector to perform the desired filtration based on the defined logic within the filterFunc.

Reply:
I mean it needs to be instantiated somewhere, right now it is always nil as it is never assigned any filter function. You would need to do that. Thats also what the linter is complaining about!

Reply:
Oh! I will have to Modify the constructor for logicalplan.VectorSelector to accept the FilterFunc as a parameter and assign it to the FilterFunc field and then pass the instantiated FilterFunc to the vector selector constructor. I will try!